### PR TITLE
developed Hud object

### LINKED ISSUE #46: Show controls during main game
There's only two, space to shoot and esc to pause. I imagine this as little button icons next to tiny text in the bottom left corner.

### LINKED ISSUE #60: Need indicators for ammunition remaining and reloading status

### LINKED ISSUE #64: Need an indicator for remaining health

### PR DESCRIPTION
**Pull request by Rw552**
resolves #46 #64 #58

### CODE REVIEW

File: game_loop.py [hunk 1]
```diff
@@ -48,6 +48,54 @@ def time_to_reload(self, game_time):
         )
 
 
+class Hud(object):
```
Comment (chwerth) [inline game_loop.py]: Now that it's done, can we move this to its own file?

File: game_loop.py [hunk 2]
```diff
@@ -156,7 +200,11 @@ def game_loop():
         # Paint the background G.WHITE
         G.SCREEN.fill(G.WHITE)
         G.SCREEN.blit(G.BACKGROUND_1.image, G.BACKGROUND_1.rect)
-        G.SCREEN.blit(scoreboard_surf, scoreboard_rect)
+        score = hud.draw_score(player.score)
```
Comment (chwerth) [inline game_loop.py]: Can we fit all this in one method? The one method (like maybe "draw") could just call all the other ones within the class

**Review by chwerth — CHANGES_REQUESTED** [review]
On my phone right now, but looks good so far. Can we also remove the __pycache__ folder?

**Comment by Rw552** [comment]
> 
> 
> On my phone right now, but looks good so far. Can we also remove the **pycache** folder?

I have no idea what that folder is for tbh.

File: game_loop.py (hunk 1 above)
Comment (Rw552) [reply game_loop.py]: yeah I'll work on that next

File: game_loop.py (hunk 2 above)
Comment (Rw552) [reply game_loop.py]: sure thing

File: game_loop.py [hunk 3]
```diff
@@ -156,7 +153,8 @@ def game_loop():
...
```
Comment (chwerth) [inline game_loop.py]: Wait, what's returned into `score`? What is this line for?

File: hud.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            "Ammo: " + ammo_status,
+            G.SMALL_TEXT,
+            G.WHITE,
+            ((G.DISPLAY_WIDTH * 0.87), (G.DISPLAY_HEIGHT * 0.93)),
+        )
+        G.SCREEN.blit(ammo_surf, ammo_rect)
+
+    def draw_controls(self):
+        control_surf, control_rect = text_objects(
+            "Press 'SPACE' to Fire!",
```
Comment (chwerth) [inline hud.py:46]: - [x] Can we move this to the bottom left? I tested this out and the top of the screen feels weird to me
- [x] Can we add `esc` to pause?
- [ ] Did you find any free spacebar/esc icons? I was thinking something like, bottom left corner, `{esc icon} pause {spacebar icon} fire` if possible

**Review by chwerth — CHANGES_REQUESTED** [review]
Okay now i'm actually at a computer. This is great so far. Good work

File: game_loop.py (hunk 3 above)
Comment (Rw552) [reply game_loop.py]: oh that's old, I can delete that.

File: hud.py (hunk 4 above)
Comment (chwerth) [reply hud.py:46]: It would also be fine to do icons in a separate, later PR to make this simpler for now

Comment (Rw552) [reply hud.py:46]: > Did you find any free spacebar/esc icons? I was thinking something like, bottom left corner, {esc icon} pause {spacebar icon} fire if possible

I couldn't find a good spacebar icon, so i opted to not use an icon for either

Comment (chwerth) [reply hud.py:46]: Aight. If we find one in the future, can just swap it in.

**Review by chwerth — APPROVED** [review]

**Comment by chwerth** [comment]
Looks good to me

**Pull request merged**